python/unitytrainers/curriculum.py

```diff
@@ -73,17 +72,17 @@ def increment_lesson(self, progress):
             progress = self.smoothing_value * 0.25 + 0.75 * progress
             self.smoothing_value = progress
         self.lesson_length += 1
-        if self.lesson_number < self.max_lesson_number:
-            if ((progress > self.data['thresholds'][self.lesson_number]) and
+        if self.lesson_num < self.max_lesson_num:
+            if ((progress > self.data['thresholds'][self.lesson_num]) and
                     (self.lesson_length > self.data['min_lesson_length'])):
                 self.lesson_length = 0
-                self.lesson_number += 1
+                self.lesson_num += 1
                 config = {}
                 parameters = self.data["parameters"]
                 for key in parameters:
-                    config[key] = parameters[key][self.lesson_number]
+                    config[key] = parameters[key][self.lesson_num]
                 logger.info("\nLesson changed. Now in Lesson {0} : \t{1}"
```

Comment:
Since there are now multiple curricula, The message should display the name of the brain that is associated. I even feel all messages should be handled by the meta_curriculum.